serving-endpoints/resource_serving_endpoint.go

```diff
@@ -0,0 +1,76 @@
+package serving_endpoints
+
+import (
+	"context"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go/retries"
+	"github.com/databricks/databricks-sdk-go/service/endpoints"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceServingEndpoint() *schema.Resource {
+	s := common.StructToSchema(
+		endpoints.CreateServingEndpoint{},
+		func(m map[string]*schema.Schema) map[string]*schema.Schema {
+			return m
+		})
+
+	return common.Resource{
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			var e endpoints.CreateServingEndpoint
+			common.DataToStructPointer(d, s, &e)
+			endpoint, err := w.ServingEndpoints.CreateAndWait(ctx, e, retries.Timeout[endpoints.ServingEndpointDetailed](45*time.Minute))
+			if err != nil {
+				return err
+			}
+			d.SetId(endpoint.Name)
+			return nil
+		},
+
+		Read: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			endpoint, err := w.ServingEndpoints.GetByName(ctx, d.Id())
+			if err != nil {
+				return err
+			}
+			return common.StructToData(*endpoint, s, d)
+		},
+
+		Update: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			var e endpoints.CreateServingEndpoint
+			common.DataToStructPointer(d, s, &e)
+			e.Config.Name = e.Name
```

Comment:
yes, this has to be done on the level of schema customization callback

Reply:
Updated!

Reply:
@alexott @nfx My `Update` test cases (both unit and acceptance) seem to be failing since I added this callback with this error.
```
2023/03/06 06:25:41 [WARN] name requires new: "" -> "test-endpoint"
```
Is there a reason it thinks that name isn't set?

Reply:
Does `read` api returns name?

Reply:
Yup it returns a [ServingEndpointDetailed](https://github.com/databricks/databricks-sdk-go/blob/main/service/endpoints/model.go#L342) object which has a [Name](https://github.com/databricks/databricks-sdk-go/blob/main/service/endpoints/model.go#L355) field.